shortfin/requirements-iree-compiler.txt

```diff
@@ -1,4 +1,4 @@
 # Keep in sync with "ref: iree-" in .github/workflows/* and GIT_TAG in CMakeLists.txt
 -f https://iree.dev/pip-release-links.html
-iree-base-compiler==3.1.0rc20241204
-iree-base-runtime==3.1.0rc20241204
+iree-base-compiler==3.1.0rc20250103
+iree-base-runtime==3.1.0rc20250103
```

Comment:
sd e2e test is failing to start the server with this change:
https://github.com/nod-ai/shark-ai/actions/runs/12604999251/job/35132935785?pr=747#step:7:408

Reply:
same as here: https://github.com/nod-ai/shark-ai/pull/722#issuecomment-2557873661

Reply:
notes to self for next week:

* Try to reproduce locally
* Have a local checkout and bisect through IREE runtime commits to find a culprit